torch/distributions/exp_family.py

```diff
@@ -52,7 +51,7 @@ def entropy(self):
         Method to compute the entropy using Bregman divergence of the log normalizer.
         """
         result = -self._mean_carrier_measure
-        nparams = [Variable(p.data, requires_grad=True) for p in self._natural_params]
+        nparams = [p.clone().requires_grad_() for p in self._natural_params]
```

Comment:
Is this equivalent? Don't we need to use something like `detach()` here?

Reply:
I am not quite sure of the semantics of `detach()`, so help would be greatly appreciated!

Reply:
@vishwakftw Ha! Then the migration guide is your best friend! See the `What about .data?` section of https://pytorch.org/2018/04/22/0_4_0-migration-guide.html .

Reply:
I actually made changes prior to the push based on the migration guide, but ran into this error, which is why I resorted to the existing version. The error is below:
```
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn
```
However, in the case with `Variable`, the older version: the `_natural_params` where not necessarily `Variable`s with `requires_grad=True`. Hence the older version.

Reply:
I think @fmassa is talking about using `.detach().requires_grad_()` combined. Did you try that too?

Reply:
Oh I see. I got it working with that. Thanks @fmassa and @SsnL